src/Chapter1.hs

```diff
@@ -640,9 +650,10 @@ Implement a function that returns the first digit of a given number.
 You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
-
-firstDigit n = error "firstDigit: Not implemented!"
-
+firstDigit :: Integral t => t -> t
+firstDigit n
+    | n < 10 = n
+    | otherwise = firstDigit (div n 10)
```

Comment:
Nice work on the advanced task! I would suggest thinking more about various corner cases, as `mod` and `div` behave differently on negative numbers.